src/datatable.py

```diff
@@ -0,0 +1,192 @@
+from collections import namedtuple
+from src import mongo
+from querystring_parser import parser
+import json
+
+max_string_length = 50
+
+
+class DataTablesServer:
+
+    def __init__(self, request, columns, index, collection, group_by=None):
+        self.columns = columns
+        self.index = index
+        self.collection = collection
+        self.group_by = group_by
+
+        # values specified by the datatable for filtering, sorting, paging
+        self.request = parser.parse(request.query_string)
+        # print(json.dumps(self.request, sort_keys=True, indent=4))
+
+        self.dbh = mongo.db
+        self.result_data = None  # results from the db
+        self.records_filtered = 0  # total in the table after filtering
+        self.records_total = 0  # total in the table unfiltered
+
+        self.run_queries()
+
+    def output_result_on_given_fields(self):
+        output = {}
+        output['draw'] = str(int(self.request['draw']))
+        output['recordsTotal'] = str(self.records_total)
+        output['recordsFiltered'] = str(self.records_filtered)
+
+        data_rows = []
+        i = self.paging().start
+
+        for row in self.result_data:
+            data_row = []
+            i += 1
+            data_row.append("<a href='/vus/{0}'>{1}</a>".format(row['_id'], i))
+            for col in self.columns:
+                if col in row:
+                    val = row[col]
+                else:
+                    val = "-"
+
+                if isinstance(val, dict):
+                    print("YAS")
+                    val = json.dumps(val)
+                    print(val)
```

Comment:
Are the print on line 48 and 50 being used? Or were those part of the development/debugging phase?

Reply:
removed them. I have checked, and it will print those statements to the commandline. I guess it was needed to check when that function would be used when clicking though the VUSualizer. So I indeed believe it was once part of an debugging phase.